shop/src/main/java/ro/msg/learning/shop/Domain/BaseClassesForIds/OrderDetailId.java

```diff
@@ -0,0 +1,18 @@
+package ro.msg.learning.shop.Domain.BaseClassesForIds;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Embeddable;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import ro.msg.learning.shop.Domain.Orders;
+import ro.msg.learning.shop.Domain.Product;
+
+import java.io.Serializable;
+import java.util.UUID;
+
+@Data
```

Comment:
So as you wrote a comment in EntityId class **@Data** contains ToString, @EqualsAndHashCode, @Getter / @Setter and @RequiredArgsConstructor together. So we need to assure that we use this annotation only when we want all sub-annotations that are contained.
Please check **here and in all other files** to only use the needed annotations from lombok.
Reference: https://projectlombok.org/features/Data